devices/Ds1302/Ds1302.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device;
+using System.Device.Gpio;
+using System.Device.Model;
+using Iot.Device.Common;
+
+namespace Iot.Device.Rtc
+{
+    /// <summary>
+    /// Realtime Clock Ds1302.
+    /// </summary>
+    [Interface("Realtime Clock Ds1302")]
+    public class Ds1302 : RtcBase, IDisposable
+    {
+        private readonly GpioController _controller;
+        private readonly GpioPin _clockPin;
+        private readonly GpioPin _resetPin;
+        private readonly GpioPin _dataPin;
+        private readonly byte enableWrite = 0b1000_0000;
+        private readonly byte disableWrite = 0b0000_0000;
+        private readonly byte enableWriteOrRead = 0b1000_0001;
+        private PinMode _currentDataPinDirection;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Ds1302"/> class.
+        /// </summary>
+        /// <param name="controller">Gpio controller.</param>
+        /// <param name="clockPin">Clock pin number.</param>
+        /// <param name="dataPin">Data pin number.</param>
+        /// <param name="resetPin">Reset pin number.</param>
+        /// <exception cref="ArgumentNullException">When controller was not provided.</exception>
+        public Ds1302(GpioController controller, int clockPin, int dataPin, int resetPin)
```

Comment:
in the pattern, we give as well a _shouldDispose. See the pattern here: https://github.com/nanoframework/nanoFramework.IoT.Device/blob/0d1d97db4da3e306349724d369fd477fa4701055/devices/A4988/A4988%20.cs#L34
And it's used in the dispose as well.
And you should then place the GpioController after the pins and have it with a default value of null. And shouldDispose to be true as default value.

Reply:
That is not going to work or would be weird because the RtcBase already implements IDisposable, so adding parameter doesn't sound like a good solution?